webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/tenants/TenantService.scala

```diff
@@ -150,4 +169,76 @@ class DefaultTenantService(var tenantsEnabled: Boolean, val tenantIds: Ref[Set[T
       } yield ns.get(nodeId).filter(qc.nodePerms.canSee(_)(ts))
     }
   }
+
+  override def manageUpdate[A](
+      existing: Option[CoreNodeFact],
+      updated:  NodeFact,
+      cc:       ChangeContext
+  )(
+      action:   NodeFact => IOResult[A]
+  ): IOResult[A] = {
+    // only id to avoid giving too much info in error in that case
+    def error(n: MinimalNodeFactInterface) = {
+      val tag = n.rudderSettings.security match {
+        case None    => '*'
+        case Some(t) => t.tenants.map(_.value).mkString(",")
+      }
+      Inconsistency(s"Node '${n.id.value}' [${tag}] can't be modified by '${cc.actor.name}' (perm:${cc.nodePerms.value})").fail
+    }
+
+    getTenants().flatMap { availableTenants =>
+      existing match {
+        // in the case of creation, we just have to check if the user has actual access on update node fact
+        case None           =>
+          if (cc.nodePerms.canSee(updated.rudderSettings.security)(availableTenants)) {
+            action(updated)
+          } else {
+            error(updated)
+          }
+        case Some(existing) =>
+          (if (cc.nodePerms.canSee(existing.rudderSettings.security)(availableTenants)) {
+             if (cc.nodePerms.canSee(updated.rudderSettings.security)(availableTenants)) {
+               // here, if tenants are not enabled, we must keep the old ones in any case
+               if (tenantsEnabled) {
```

Comment:
Maybe it's kinda lata, but does the tenantEnabled should have been a private ref inside of the service ? here anything from outside the service can modify it, or access it asynchronously ?